File: csharp/Library/Member.cs
namespace SammanCoaching.Library;
public sealed record Member
{
    public Member(string memberId, string name)
    {
        MemberId = RequireText(memberId, nameof(memberId));
        Name = RequireText(name, nameof(name));
    }
    public string MemberId { get; }
    public string Name { get; }

    private static string RequireText(string value, string fieldName)
    {
        ArgumentNullException.ThrowIfNull(value, fieldName);
        if (string.IsNullOrWhiteSpace(value))
        {
            throw new ArgumentException($"{fieldName} must not be blank", fieldName);
        }

        return value;
    }
}

File: java/lib/src/main/java/org/sammancoaching/library/Member.java
package org.sammancoaching.library;

import java.util.Objects;
public record Member(String memberId, String name) {
    public Member {
        memberId = requireText(memberId, "memberId");
        name = requireText(name, "name");
    }

    private static String requireText(String value, String fieldName) {
        Objects.requireNonNull(value, fieldName);
        if (value.isBlank()) {
            throw new IllegalArgumentException(fieldName + " must not be blank");
        }
        return value;
    }
}
